docs/csharp/reference-semantics-with-value-types.md

```diff
@@ -202,6 +202,20 @@ types.
 These restrictions ensure that you do not accidentally use a `ref struct`
 in a manner that could promote it to the managed heap.
 
+## `readonly ref struct` type
+
+Declaring a struct as `readonly ref` combines the benefits and restrictions of `ref struct` and `readonly struct` delcarations. 
+
+The following example demonstrates the declaration of `readonly ref struct`.
+
+```csharp
```

Comment:
We need a better example here. I'd suggest adding the following constructor:

```csharp
ReadOnlyRefPoint2D(int x, int y) => (X, Y) = (x, y);
```

Otherwise, `X` and `Y` can only have the value 0.